### PR TITLE
[stable/memcached] Bump memcached version, add Kubernetes Service type configuration.

### PR DESCRIPTION
**Pull request by janghwan**
Added Kubernetes Service type configuration for the memcached chart.
Signed-off-by: Janghwan <cheesespy@gmail.com>

#### Which issue this PR fixes
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
  - fixes #

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] [DCO](https://github.com/helm/charts/blob/master/CONTRIBUTING.md#sign-your-work) signed
- [x] Chart Version bumped
- [x] Variables are documented in the README.md
- [x] title of the PR contains starts with chart name e.g. `[stable/chart]`

### CODE REVIEW

**Comment by olemarkus** [comment]
/assign

Why would you want to change the service type? If you put multiple memcached behind a load balancer, you'll get inconsistent hit/miss/cache updates depending on what server you hit.

**Comment by janghwan** [comment]
> /assign
> 
> Why would you want to change the service type? If you put multiple memcached behind a load balancer, you'll get inconsistent hit/miss/cache updates depending on what server you hit.

I need to access Memcached from outside of the kubernetes cluster. I had to set up only one replica due to the issue you mentioned.
The idea of exposing service with the service type is from the bitnami helm chart.

File: stable/memcached/templates/svc.yaml [hunk 1]
```diff
@@ -10,7 +10,7 @@ metadata:
   annotations:
 {{ toYaml .Values.serviceAnnotations | indent 4 }}
 spec:
-  clusterIP: None
+  type: {{ .Values.serviceType }}
```
Comment (olemarkus) [inline stable/memcached/templates/svc.yaml:14]: You cannot remove this line. That would make the service non-headless.

File: stable/memcached/Chart.yaml [hunk 2]
```diff
@@ -1,6 +1,6 @@
 apiVersion: v1
 name: memcached
-version: 2.8.2
+version: 2.8.3
```
Comment (olemarkus) [inline stable/memcached/Chart.yaml]: You need to bump the minor version here, as a new feature is being added.

**Review by olemarkus — CHANGES_REQUESTED** [review]

**Pull request closed without merging**